src/webui/www/private/scripts/client.js

```diff
@@ -843,6 +843,21 @@ window.addEvent('load', function() {
         else
             document.title = ("qBittorrent " + qbtVersion() + " QBT_TR(Web UI)QBT_TR[CONTEXT=OptionsDialog]");
         $('freeSpaceOnDisk').set('html', 'QBT_TR(Free space: %1)QBT_TR[CONTEXT=HttpServer]'.replace("%1", window.qBittorrent.Misc.friendlyUnit(serverState.free_space_on_disk)));
+
+        let last_external_address_v4 = serverState.last_external_address_v4;
+        let last_external_address_v6 = serverState.last_external_address_v6;
+        let last_external_addresses = 'QBT_TR(External Address(es): Detecting)QBT_TR[CONTEXT=HttpServer]';
+
+        if (last_external_address_v4 !== "" || last_external_address_v6 !== "")
+        {
+            if (last_external_address_v4 !== "" && last_external_address_v6 !== "")
+                last_external_addresses = 'QBT_TR(External Addresses: %1, %2)QBT_TR[CONTEXT=HttpServer]';
```

Comment:
Although unlikely, I would still suggest using other character for replacement specifier. Maybe `$`
See: https://en.wikipedia.org/wiki/IPv6_address#Scoped_literal_IPv6_addresses_(with_zone_index)

Reply:
Unfortunately, according to the linked article, Windows will use %[0-9] format:
>The latter (using an interface number) is the standard syntax on Microsoft Windows ...

Reply:
I can see the interface parts in qbt under Windows.

Reply:
>This is not at all about getting the local IP address even if it is your external, this is about reporting "what the world sees" as your external address (which will probably be your local address if it's a VPN/VPS/local installation with a public IP address).

This is why I said it is "unlikely" in my [first post](https://github.com/qbittorrent/qBittorrent/pull/20118/files#r1428698907). Surely using a different placeholder character that would avoid all potential problems isn't a bad idea or is even hard, no?

Reply:
>And no, Qt doesn't support another parsing character besides percent.

What Qt? Here we are talking about javascript.

Reply:
>"QBT_TR" ... The backend is Qt and this part uses Qt parsing, for localization and such. Plus the code exists in Qt as well for the non-nox users.

That 'special annotation' has nothing to do with `%` or `$` or any character used for replacement purpose.

I'll just provide the code:
```js
last_external_addresses = 'QBT_TR(External Addresses: $1, $2)QBT_TR[CONTEXT=HttpServer]';
// ...
$('externalAddresses').set('html', last_external_addresses.replace("$1", last_external_address_v4).replace("$2", last_external_address_v6));
```
See the `.replace()`? here is the documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

Reply:
>Plus the code exists in Qt as well for the non-nox users.

I don't intend you to change that. GUI, WebUI have their own set of translation strings and here I'm only talking about WebUI.

Reply:
> It literally makes a web request to a remote server (think "what's my ip address" in a search engine query) and returns the result.

`libtorrent` gets the external addresses from trackers and peers. This is clearly stated in the documentation (and I can confirm this based on an inspection of the `libtorrent` code).

Reply:
> Still, there isn't any way for a peer or tracker to know which interface the client is using,

Sorry, I don't know much about IPv6. What exactly are you talking about when you mention the interface? About the zone identifier, e.g. `%eth0`?

Reply:
> the IPv6 address might have the zone identifier reported

IIRC, Zone ID is "locally significant" (enables us to define out which interface we want to send some traffic) and there is no point to send it to remote host as part if its "external address". But we still can't be sure that it isn't sent in such a way (with Zone ID) so the question is how is it handled by libtorrent, i.e. can it provide we "external address" that mistakenly contains such Zone ID.